page_iterator.go

```diff
@@ -0,0 +1,195 @@
+package msgraphgocore
+
+import (
+	"log"
+	"net/url"
+	"reflect"
+	"unsafe"
+
+	abstractions "github.com/microsoft/kiota/abstractions/go"
+	"github.com/microsoft/kiota/abstractions/go/serialization"
+	jsonserialization "github.com/microsoft/kiota/serialization/go/json"
+)
+
+type Page interface {
+	getValue() []interface{}
+	getNextLink() *string
+}
+
+// PageIterator represents an iterator object that can be used to get subsequent pages of a collection.
+type PageIterator struct {
+	currentPage     Page
+	reqAdapter      GraphRequestAdapterBase
+	pauseIndex      int
+	constructorFunc ParsableConstructor
+	headers         map[string]string
+}
+
+type ParsableConstructor func() serialization.Parsable
```

Comment:
what do you think about moving this to kiota abstractions so we can use it in the other interfaces, the http and the json libs?

Reply:
Not familiar with Kiota abstractions yet, so I trust your judgement here.

Reply:
no worries, please go ahead and open a PR over there to implement that change, we'll merge it ahead of type one so you can take advantage of it and avoid duplication.

Reply:
unresolving this as this is not completed yet.

Reply:
Yeah, I didn't see the point since **ParsableConstructor** is an alias (to improve readability) of a type that is already in kiota abstraction.

Reply:
alright, that alias could also improve readability in kiota packages but I understand if you want to do that at a later time. resolving.